p2p/host/peerstore/pstoreds/addr_book.go

```diff
@@ -601,10 +601,12 @@ func (ab *dsAddrBook) deleteAddrs(p peer.ID, addrs []ma.Multiaddr) (err error) {
 func cleanAddrs(addrs []ma.Multiaddr) []ma.Multiaddr {
```

Comment:
This should now take a peer id as well, and filter out multiaddrs that have an id component but don't match (and log an error when this happens)

Reply:
fixed now

Reply:
I'm logging a warning which was previously logged for nil addresses, should I make it an error?